production/ksonnet/README.md

```diff
@@ -54,19 +57,36 @@ promtail + {
 ```
 Notice that `container_root_path` is your own data root for docker daemon, use `docker info | grep "Root Dir"` to get it.
 
-Then do `ks show loki` to see the manifests that'll be deployed to your cluster.
-Apply them using `ks apply loki`.
+As a last step, fill add the correct `spec.apiServer` and `spec.namespace` to `environments/default/spec.json`:
+
+{
+  "apiVersion": "tanka.dev/v1alpha1",
+  "kind": "Environment",
+  "spec": {
+    "apiServer": "https://localhost:6443",
+    "namespace": "default"
+  }
+}
+
+Now use `tk show environments/default` to see the yaml, and `tk apply environments/default` to apply it to the cluster.
 
 ## Deploying Loki to your cluster.
 
 If you want to further also deploy the server to the cluster, then run the following to install the module:
 
 ```
-jb install github.com/grafana/loki/production/ksonnet/loki
+# create a directory (any name works)
+$ mkdir loki && cd loki/
+$ tk init
+$ tk env add loki --namespace=loki
+# Ksonnet kubernetes libraries
+$ jb install github.com/ksonnet/ksonnet-lib/ksonnet.beta.3/k.libsonnet
+$ jb install github.com/ksonnet/ksonnet-lib/ksonnet.beta.3/k8s.libsonnet
+$ jb install github.com/grafana/loki/production/ksonnet/loki
 ```
 
 Be sure to replace the username, password and the relevant htpasswd contents.
-Replace the contents of `environments/loki/main.jsonnet` with:
+Replace the contents of `environments/default/main.jsonnet` with:
```

Comment:
`s/default/loki/`

Reply:
I have a question about this:
first i download promtail  and do change to main/spec, then i download loki still change to main/spec, the files(main/spec) always overvide by download?

Reply:
No, `jb` NEVER touches `spec.json` nor `main.jsonnet`. All it does is placing libraries into the `vendor/` folder, to be imported by the user from Jsonnet.